mdal/mdal_defines.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ MDAL - Mesh Data Abstraction Library (MIT License)
+ Copyright (C) 2018 Peter Petrik (zilolv at gmail dot com)
+*/
+
+#include <mdal_defines.hpp>
+#include <assert.h>
+
+void MDAL::Dataset::free()
```

Comment:
The free() method is confusing... why this is not a destructor? (the same for DatasetGroup)